extensions/window/init.lua

```diff
@@ -682,21 +681,28 @@ end
 ---    - an `hs.geometry` unit rect, or argument to construct one; will set the window frame to this rect relative to the desired screen;
 ---      if `screen` is nil or omitted, use the screen the window is currently on
 ---  * screen - (optional) An `hs.screen` object or argument for `hs.screen.find`; only valid if `rect` is a unit rect
+---  * ensureInScreenBounds - (optional) if `true`, use `setFrameInScreenBounds()` to ensure the resulting window frame is fully contained within
+---    the window's screen; `duration` is ignored (animation is not supported)
 ---  * duration - (optional) A number containing the number of seconds to animate the transition. Defaults to the value of `hs.window.animationDuration`
 ---
 --- Returns:
 ---  * The `hs.window` object
 function window:move(rect,toScreen,duration)
+  local frame,inBounds
+  if type(toScreen)=='boolean' or (type(toScreen)=='number' and toScreen<20 and duration==nil) then duration=toScreen toScreen=nil end
```

Comment:
I'm going to firmly -1 this <20 thing. It says to me that we have gone too far down the road of accepting arguments in multiple formats. If we need an hs.screen object, the user should supply an hs.screen object. Trying to do clever dances to save them typing hs.screen.find(blah) is causing us more hassle than it's worth, IMO.